slacksite/slackbot/views.py

```diff
@@ -11,6 +14,42 @@ def posting(request):
     user_id = r.get('user_id')
     user_name = r.get('user_name')
     text = r.get('text')
-    insert_user_db(user_id, user_name)
+    new_user, created = User.objects.get_or_create(user_id=user_id, 
+        user_name=user_name)
+    new_user.save()
     update_offence_db(user_id, text)
     return HttpResponse(str(request))
+
+
+def get_search_param(req):
+    req = str(req)
+    req = req.replace('>', "").replace("'", "").replace('>', "")
+    req = req.split('/')
+    return req[len(req) - 1]
+
+
+def leaderboard(request):
+    user_list = OffenceLog.objects.values('user').distinct()
+    data_span = get_search_param(request)
+    enddate = date.today()
+    what_to = enddate.weekday()
+    if data_span == "monthly":
+        startdate = enddate.replace(day=1)
+    else:
+        startdate = enddate - timedelta(days=what_to)
+    leader = ""
+    for user in user_list:
+        name_obj = User.objects.get(user_id=user['user'])
+        on_leave = OffenceLog.objects.all().filter(user=user['user'],
+            offence_type="on_leave", timestamp__range=[startdate, enddate])
+        arrive_late = OffenceLog.objects.all().filter(user=user['user'], offence_type="come_late")
+        leave_early = OffenceLog.objects.all().filter(user=user['user'], offence_type="leaving_early")
+        leader += ('<tr style="color:blue;">' +
```

Comment:
You have to return a JSON response. And using that response, you have to show the data on UI.